topdown/sets.go

```diff
@@ -23,6 +25,48 @@ func builtinSetDiff(a, b ast.Value) (ast.Value, error) {
 	return s1.Diff(s2), nil
 }
 
+// builtinSetIntersection returns the intersection of the given input sets
+func builtinSetIntersection(a ast.Value) (ast.Value, error) {
+	switch a := a.(type) {
```

Comment:
You can use builtins.SetOperand here to cast a to a set or return an error if cast fails. Same thing below in Union and in the accumulators.

Reply:
Done